src/main/java/uk/gov/hmcts/sscs/service/SubmitAppealService.java

```diff
@@ -85,50 +85,63 @@
     }
 
     public void submitAppeal(SyaCaseWrapper appeal) {
-        //add RPC and venue
         String postcode = getFirstHalfOfPostcode(appeal.getAppellant().getContactDetails().getPostCode());
+
+        CaseData caseData = prepareCaseForCcd(appeal, postcode);
+        CaseDetails caseDetails = createCaseInCcd(caseData);
+
+        byte[] pdf = generatePdf(appeal, caseDetails.getId(), caseData);
+
+        prepareCaseForPdf(appeal, caseDetails.getId(), caseData, pdf);
+
+        if (roboticsEnabled) {
+            sendCaseToRobotics(appeal, caseDetails.getId(), postcode, pdf);
+        }
+    }
+
+    private CaseData prepareCaseForCcd(SyaCaseWrapper appeal, String postcode) {
         String region = airLookupService.lookupRegionalCentre(postcode);
-        String venue = airLookupService.lookupAirVenueNameByPostCode(postcode);
         RegionalProcessingCenter rpc = regionalProcessingCenterService.getByName(region);
 
         CaseData caseData;
+
         if (rpc == null) {
             caseData = transformAppealToCaseData(appeal);
         } else {
             caseData = transformAppealToCaseData(appeal, rpc.getName(), rpc);
         }
 
-        CaseDetails caseDetails = createCaseInCcd(caseData);
-
-        log.info("Appeal successfully created in CCD for  Nino - {} and benefit type {}",
-                appeal.getAppellant().getNino(), appeal.getBenefitType().getCode());
-
-        byte[] pdf = generatePdf(appeal, caseDetails.getId(), caseData);
+        return caseData;
```

Comment:
I think this might be a bit cleaner, Jack:
if (rpc == null) {
return transformAppealToCaseData(appeal);
} else {
return transformAppealToCaseData(appeal, rpc.getName(), rpc);
}